File: src/blockchain_handler/zmq_ipc_examples/worker.py
import sys
import os
import time
import zmq
import random

def consumer():
    consumer_id = random.randrange(1,10005)
    print "I am consumer #%s" % (consumer_id)
    sys.stdout.flush()
    context = zmq.Context()

    # recieve work
    consumer_receiver = context.socket(zmq.PULL)
    consumer_receiver.connect(os.environ["SOCKET"])

    while True:
        work = consumer_receiver.recv()
        print "[>>] python worker: " + work
        sys.stdout.flush()

consumer()

File: src/blockchain_handler/zmq_ipc_examples/worker.js
var zmq = require('zmq')
  , sock = zmq.socket('pull');

sock.connect(process.env.SOCKET);
console.log('Worker connected to socket %s', process.env.SOCKET);

sock.on('message', function(msg){
  console.log('[>>] nodejs worker: %s', msg.toString());
});
